src/test/java/uk/gov/pay/ledger/transaction/resource/TransactionResourceIT.java

```diff
@@ -190,6 +191,84 @@ public void shouldReturnTransactionSearchResponseObjectForNonExistentTransaction
                 .body("results.size()", is(0));
     }
 
+    @ParameterizedTest
+    @CsvSource({"?, %3F",
+            "{ , %7B",
+            "[ , %5B",
+            "f{o{o}{ , f%7Bo%7Bo%7D%7B",
+            "foo&, foo%26",
+            "foo@ , foo%40",
+            "foo@@ , foo%40%40",
+            "foo=bar&baz=quux , foo%3Dbar%26baz%3Dquux",
+            "foo bar , foo+bar",
+            "% , %25",
+            "%7B, %257B"
+    })
+    public void shouldReturnTransactionSearchResponseObjectForExistingTransactionWhenReferenceIsUrlEncoded(String rawValue, String encodedValue) {
+        transactionFixture = aTransactionFixture()
+                .withReference(rawValue);
+        transactionFixture.insert(rule.getJdbi());
+        
+        given().port(port)
+                .contentType(JSON)
+                .accept(JSON)
+                .urlEncodingEnabled(false)
+                .get("/v1/transaction" +
+                        "?reference=" +
+                        encodedValue +
+                        "&account_id=" + transactionFixture.getGatewayAccountId()
+                )
+                .then()
+                .statusCode(Response.Status.OK.getStatusCode())
+                .contentType(JSON)
+                .body("results[0].reference", is(rawValue));
+    }
+    
+    @Test
+    public void shouldReturnATransactionSearchResponseObjectBetweenTwoDates() {
+        var fromDate = ZonedDateTime.parse("2023-11-01T00:00:00.000Z");
+        var toDate = ZonedDateTime.parse("2023-12-01T00:00:00.000Z");
+        var dateBetween = ZonedDateTime.parse("2023-11-15T14:52:07.073Z");
```

Comment:
I think these should all work as `Instant`s (we generally prefer `Instant`s to `ZonedDateTime`s because time zones make things complicated).